main/pipeline/pipeline.cc

```diff
@@ -578,39 +578,41 @@ ast::ParsedFilesOrCancelled mergeIndexResults(core::GlobalState &cgs, const opti
         Timer timeit(cgs.tracer(), "substituteTrees");
         auto resultq = make_shared<BlockingBoundedQueue<vector<ast::ParsedFile>>>(batchq->bound);
 
-        workers.multiplexJob("substituteTrees", [&cgs, &logger = cgs.tracer(), batchq, resultq, cancelable]() {
-            Timer timeit(logger, "substituteTreesWorker");
-            IndexSubstitutionJob job;
-            int numTreesProcessed = 0;
-            std::vector<ast::ParsedFile> trees;
-            for (auto result = batchq->try_pop(job); !result.done(); result = batchq->try_pop(job)) {
-                if (result.gotItem()) {
-                    // Unconditionally update the total to avoid starving the consumer thread
-                    numTreesProcessed += job.numTreesProcessed;
+        workers.multiplexJob(
+            "substituteTrees", [&cgs = as_const(cgs), &logger = cgs.tracer(), batchq, resultq, cancelable]() {
```

Comment:
I am on my phone, so it's hard to be 100% sure, but it looks like we don't mutate `GlobalState` in this function, so we could just take it as a `const` in this function from the start?

Reply:
I think that the earlier block is using `cgs` mutably (I think the earlier block is where we actually merge the GlobalStates.

That's probably worth another look just to be sure, but I think we can't change `mergeIndexResults` to take a `const GlobalState &`